File: MoviesApplication/Data/Models/Director.cs
using System.ComponentModel.DataAnnotations;

namespace MoviesApplication.Data.Models
{
    public class Director
    {
        [Key]
        public Guid Id { get; set; }
        public string FirstName { get; set; }
        public string SecondName { get; set; }
        public DateTime? Birthday { get; set; }
        public string? CountryOfBirth { get; set; }
        public string? PhotoUrl { get; set; }
        public ICollection<Movie>? Movies { get; set; }
    }
}

File: MoviesApplication/ClientApp/src/app/shared/director.ts
import { Movie } from './movie';

export default class Director {
  id!: string;
  firstName!: string;
  secondName!: string;
  birthday!: Date | null;
  countryOfBirth!: string | null;
  photoUrl!: string | null;
  movies!: Movie[] | null;
}
